guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,6 @@
+## Performance related
+
+* Avoid allocations in hot paths:
+* Avoid LINQ.
+* Avoid using foreach over collections that do not have a struct enumerator. For public facing interfaces we favor readonly collections and enumerables. For internal types we favor speed and allocation reduction, therefore avoid using collection interfaces internally.
```

Comment:
That part was just moved from the contributors.md but please go ahead and raise a issue!